src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -117,7 +117,15 @@ SendBareCommandListToWorkers(TargetWorkerSet targetWorkerSet, List *commandList)
 		int nodePort = workerNode->workerPort;
 		int connectionFlags = FORCE_NEW_CONNECTION;
 
-		if (targetWorkerSet == WORKERS_WITH_METADATA && !workerNode->hasMetadata)
+		if ((targetWorkerSet == WORKERS_WITH_METADATA ||
```

Comment:
Ops, this approach looks wrong :/  Workers may not know about the other workers. See #2320. 

I think we need to work on this again. We might consider the initial idea where any node issues something like `LOCK TABLE IF EXISTS x IN EXLUSIVE MODE ` to all other nodes.

And, we should never have something like `OTHER_WORKERS_WITH_METADATA `